### ISSUE TITLE
[Android] Wrong quick action triggered

### ISSUE THREAD

**Issue by JakeHedman** [notifee/react-native-notifee#359]
I have three actions on my notification and for about 1/20 taps the next action is triggered instead of the tapped one (second instead of first, third instead of second).

Notification:
```js
notifee.displayNotification({
  title: 'title',
  body: 'body',
  android: {
    channelId: 'test',
    actions: [
      {
        title: 'First',
        pressAction: {
          id: 'first',
        },
      },
      {
        title: 'Second',
        pressAction: {
          id: 'second',
        },
      },
      {
        title: 'Third',
        pressAction: {
          id: 'third',
        },
      },
    ],
  },
})
```

Notifee 1.10.1
React native 63

**Comment by mikehardy** [comment]
That is most unexpected! Do you have a platform / model / version combo that this shows up on, or is it all of them?

e.g. "reproduced on Android 10 OnePlus 6T" or "shows up on every device I've tried, including specifically Android emulators with debug builds, ios simulators in release only" etc

I ask because this *may* be a race condition thing and be dependent on execution environment, so it may help to know where you definitely see it, allowing us to pinpoint more quickly

Thanks!

**Comment by JakeHedman** [comment]
> That is most unexpected! Do you have a platform / model / version combo that this shows up on, or is it all of them?
> 
> e.g. "reproduced on Android 10 OnePlus 6T" or "shows up on every device I've tried, including specifically Android emulators with debug builds, ios simulators in release only" etc
> 
> I ask because this _may_ be a race condition thing and be dependent on execution environment, so it may help to know where you definitely see it, allowing us to pinpoint more quickly
> 
> Thanks!

It is indeed reproduced on Android 10 OnePlus 6T, and on an Android 11 emulator.

**Comment by mikehardy** [comment]
Nice choice of device (I picked that arbitrarily but it's my real phone ;-) ) - Always easier when it's available on an emulator in debug though. Thanks for the extra information

**Comment by helenaford** [comment]
@JakeHedman which event handler is getting triggered? is it the `getInitialNotification`, `onBackgroundEvent` or `onForegroundEvent` or all of them?

**Comment by JakeHedman** [comment]
> @JakeHedman which event handler is getting triggered? is it the `getInitialNotification`, `onBackgroundEvent` or `onForegroundEvent` or all of them?

At the moment I can only reproduce this locally with `onForegroundEvent`, but bug reports from the wild shows that it is happening from notifications when the app is closed too. The notification in question shows up when work starts and it's most likely the app has been closed for ~15 hours since the last work day ended.

We received a screen recording from a customer showing how option 2 being tapped, but the action for option 3 was triggered. In our testing we can see that we get the wrong action ID from notifee sometimes so we think that is the root cause here.

FYI we're using `onBackgroundEvent` to get the signal to display the notification in the first place.

**Comment by helenaford** [comment]
OK thanks for the info, I will look to see if I can reproduce it. Are there specific steps you do to trigger it, or is it simply display a bunch of notifications and then press an action? Do you have to press an action one after the other in quick succession?

**Comment by mikehardy** [comment]
I was thinking have the handler register which action was pressed then display that as a new notification with a fresh set of action buttons :-). Could turn this into an ad-supported game perhaps

**Comment by JakeHedman** [comment]
> OK thanks for the info, I will look to see if I can reproduce it. Are there specific steps you do to trigger it, or is it simply display a bunch of notifications and then press an action? Do you have to press an action one after the other in quick succession?

Yup, just a simple display and press an action! I do not have to press an action after the other or something like that, just one time will do (which is why our customers are complaining about it :laughing:).

If I tap an action button multiple times on the same notification it will trigger the correct action each time or the wrong action each time, so it seems like something is messed up on the notification and not just the individual action press. On the broken notifications action button one will trigger two, and two will trigger three, three will trigger three.

**Comment by mikehardy** [comment]
Your hunch is probably correct in your second paragraph, thanks again for all the repro information

**Comment by JakeHedman** [comment]
This also happens in version 0.12.1 btw!

**Comment by Stephane-Parent** [comment]
Hi, we use Notifee 1.4 here and discover this issue recently. It happens on Android 10 (Samsung S9) quite often. But cannot reproduce on Android 8 (Samsung S7) even with many very quick taps. 

Any progress on that fix?

**Comment by JakeHedman** [comment]
Minimal example to reproduce: https://github.com/JakeHedman/notifeeWrongActionButton

It's just a fresh `react-native init` project with some code to show a notification in App.js. Press the red area to display a notification, then tap "first" to trigger a console.log. It should log "first", but once every blue moon it will log "second".

I use this command to keep tapping automatically: `while true; do adb shell input tap 90 300; sleep 0.5; done`
